drivers/wifi/eswifi/eswifi_socket.c

```diff
@@ -144,17 +153,16 @@ static void eswifi_off_read_work(struct k_work *work)
 	}
 
 	if (!len || !socket->recv_cb) {
+		net_pkt_unref(pkt);
 		goto done;
 	}
 
-	LOG_DBG("payload sz = %d", len);
-
-	pkt = net_pkt_rx_alloc_with_buffer(eswifi->iface, len,
-					   AF_UNSPEC, 0, K_NO_WAIT);
-	if (!pkt) {
-		LOG_ERR("Cannot allocate rx packet");
-		goto done;
+	/* Resize the packet */
```

Comment:
Instead of this, you can call `net_pkt_compact` that should remove empty net_buf from net_pkt. This should be called after net_pkt_write.

Reply:
Is it not better to use net_pkt_trim_buffer() after net_pkt write ? It checks for unused buffers and deallocate them relevantly.

Reply:
Trim is ok too, I just forgot it existed